application/src/test/java/uk/nhs/hee/tis/revalidation/controller/RecommendationControllerTest.java

```diff
@@ -78,6 +87,14 @@ class RecommendationControllerTest {
   @MockBean
   private RecommendationService service;
 
+  @Captor
+  ArgumentCaptor<BindingResult> bindingResultArgumentCaptor;
+
+  @MockBean
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @SpyBean
```

Comment:
This is a misuse of a `@SpyBean`.  The [Mockito Docs](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#13) discourage use where there are alternatives.
I may have missed why a `@MockBean` doesn't work.

Reply:
When we tried `@MockBean` and many tests were failing, we were looking for a solution and that is. `@SpyBean ` which works here well.

Reply:
I think the issue here was that validate is a void method that uses a "pass by reference" implementation to populate the errors variable - which makes it hard to mock. 

Alternatives are we can write a stubbing using [doAnswer](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#doAnswer(org.mockito.stubbing.Answer)) for each method in the class, or refactor the validate method to return the error value

Reply:
I think `@SpyBean` here does some trick like `@Autowired` (and it works for injection), but it doesn't really spy on the validator.